### PR TITLE
911 Rework: Moved Buttons, Police -> Marshals, Confirmation Prompts, SWAT Rework, SolFed Military Fixed Guard Trim Sprites

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

All 911 calls now require you to go through two confirmation prompts.
These prompts clarify what you should/should not call 911 for.
All 911 calls now require you to type a reason for calling 911.
AI and Cyborgs can no longer call 911.

Police have been renamed to Marshals and given cowboy outfits.
All First Responders, SWAT, and Military now have cell phones to communicate. This is to reduce overhead on editing radio code while providing Sol Federation responders with a method to communicate.
Sol Federation ID cards have had their access fixed, and all responder IDs now properly say their name.

Only 911 Responders can use Sol Federation devices such as the Beamout Tool, Faulty 911 Call Reporter, and Treason Reporter.

100% of the First Responder team will need to vote for a Faulty Call now instead of 50%.

All First Responders and SWAT teams can now use their assigned Beamout Tool to return to Sol Federation space along with whoever they've got in tow. This will allow them to leave the station on their own without administrative intervention, along with anyone they've arrested, if they so choose.

SWAT teams can now declare the station as having committed Treason, following four or five confirmation prompts and an acceptance that misuse of this feature can and will result in administrative action.

## How This Contributes To The Skyrat Roleplay Experience
Improves the 911 feature to function better, with less confusion and more cooperation between first responders and SWAT teams.

Patches an issue where the station thought that shooting at the government had no consequences.
## Changelog

:cl:
balance: All 911 calls now require you to go through two confirmation prompts. These prompts clarify what you should/should not call 911 for.
balance: All 911 calls now require you to type a reason for calling 911.
balance: AI and Cyborgs can no longer call 911.
fix: 911 buttons have been moved to the bottom of the list to remove accidental presses.
balance: Police have been renamed to Marshals and given cowboy outfits.
add: All First Responders, SWAT, and Military now have cell phones to communicate. This is to reduce overhead on editing radio code while providing Sol Federation responders with a method to communicate.
fix: Sol Federation ID cards have had their access fixed, and all responder IDs now properly say their name.
fix: Only 911 Responders can use Sol Federation devices such as the Beamout Tool, Faulty 911 Call Reporter, and Treason Reporter.
balance: 100% of the First Responder team will need to vote for a Faulty Call now instead of 50%.
add: All First Responders and SWAT teams can now use their assigned Beamout Tool to return to Sol Federation space along with whoever they've got in tow. This will allow them to leave the station on their own without administrative intervention, along with anyone they've arrested, if they so choose.
add: SWAT teams can now declare the station as having committed Treason, following four or five confirmation prompts and an acceptance that misuse of this feature can and will result in administrative action, along with a full team vote.
fix: Fixes the trim visuals on the Departmental Guard IDs.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -411,15 +411,35 @@ GLOBAL_VAR_INIT(cops_arrived, FALSE)
...
+				return
+			message_admins("[ADMIN_LOOKUPFLW(usr)] is considering calling the Sol Federation Marshals.")
+			if(tgui_input_list(usr, "Are you sure you want to call 911? Faulty 911 calls results in a $20,000 fine and a 5 year superjail \
+			sentence.", "Call 911", list("Yes", "No")) != "Yes")
+				return
+			message_admins("[ADMIN_LOOKUPFLW(usr)] has acknowledged the faulty 911 call consequences.")
+			if(tgui_input_list(usr, "You SHOULD call Marshals for:\n\
+			Security ignoring Command, Security violating civil rights, Security engaging in Mutiny, General Violation of Sol Federation Citizen Rights by Command/Security, etc.\n\
+			You SHOULD NOT call Marshals for:\n\
+			Corporate affairs, antagonist hunting, settling arguments, etc.\n\
```
Comment (GoldenAlpharex) [inline code/game/machinery/computer/communications.dm:425]: I wouldn't literally say "Antagonist hunting" lmao

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -411,15 +411,35 @@ GLOBAL_VAR_INIT(cops_arrived, FALSE)
...
+				return
 			GLOB.cops_arrived = TRUE
-			send_in_the_fuzz(EMERGENCY_RESPONSE_POLICE)
+			GLOB.call_911_msg = reason_to_call_911
 			GLOB.caller_of_911 = usr.name
-			to_chat(usr, span_notice("Authorization confirmed. 911 call dispatched to the Sol Federation Police Department."))
-			playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, FALSE)
+			log_game("[key_name(usr)] has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]")
+			message_admins("[ADMIN_LOOKUPFLW(usr)] has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]")
+			deadchat_broadcast(" has called the Sol Federation Marshals for the following reason:\n[GLOB.call_911_msg]", span_name("[usr.real_name]"), usr, message_type=DEADCHAT_ANNOUNCEMENT)
```
Comment (GoldenAlpharex) [inline code/game/machinery/computer/communications.dm:438]: I assume that's how deadchat_broadcast works, even if it looks reversed.

File: code/game/machinery/computer/communications.dm [hunk 3]
```diff
@@ -411,15 +411,35 @@ GLOBAL_VAR_INIT(cops_arrived, FALSE)
...
 
-			log_game("[key_name(usr)] has called the Sol Federation Police Department.")
-			message_admins("[ADMIN_LOOKUPFLW(usr)] has called the Sol Federation Police Department.")
-			deadchat_broadcast(" has called the Sol Federation Police Department.", span_name("[usr.real_name]"), usr, message_type=DEADCHAT_ANNOUNCEMENT)
+			send_in_the_fuzz(EMERGENCY_RESPONSE_POLICE)
```
Comment (GoldenAlpharex) [inline code/game/machinery/computer/communications.dm:440]: I'd still like a better name for this proc. It might mean something to you, but I don't know what a "fuzz" is and that's never something I'd search for.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm [hunk 4]
```diff
@@ -1,7 +1,11 @@
 GLOBAL_VAR_INIT(amt_911_responders, 0)
 GLOBAL_VAR_INIT(votes_for_bad_911_call, 0)
+GLOBAL_VAR_INIT(amt_SWAT_responders, 0)
+GLOBAL_VAR_INIT(votes_for_treason, 0)
 GLOBAL_VAR_INIT(fradulent_911_declared, FALSE)
+GLOBAL_VAR_INIT(treason_declared, FALSE)
 GLOBAL_VAR(caller_of_911)
+GLOBAL_VAR(call_911_msg)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/sol_fed.dm:8]: Can we really not do without **9** global variables? I'm sure there's a better way to do this.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm [hunk 5]
```diff
@@ -33,51 +37,77 @@ GLOBAL_LIST_INIT(emergency_responders, list())
...
+
+	var/obj/item/card/id/C = H.wear_id
+	if(istype(C))
+		shuffle_inplace(C.access) // Shuffle access list to make NTNet passkeys less predictable
+		C.registered_name = H.real_name
+		if(H.age)
+			C.registered_age = H.age
+		C.update_label()
+		C.update_icon()
+		var/datum/bank_account/B = SSeconomy.bank_accounts_by_id["[H.account_id]"]
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/sol_fed.dm:80]: Why do they have a bank acount? Is this just some copy-paste of code that's already somewhere else? Why is there so many single-letter vars?

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm [hunk 6]
```diff
@@ -158,17 +194,65 @@ GLOBAL_LIST_INIT(emergency_responders, list())
...
+/datum/antagonist/ert/request_911/treason_destroyer/greet()
+	var/missiondesc =  ""
+	missiondesc += "<B><font size=5 color=red>You are NOT a Nanotrasen Employee. You work for the Sol Federation as a [role].</font></B>"
+	missiondesc += "<BR>You are here to assume control of [station_name()] due to the occupants engaging in Treason as reported by our SWAT team.\n"
+	missiondesc += "<BR><B>Your Mission</B>:"
+	missiondesc += "<BR> <B>1.</B> Contact the SWAT Team and the First Responders via your cell phone to get the situation from them."
+	missiondesc += "<BR> <B>2.</B> Arrest all suspects involved in the treason attempt."
+	missiondesc += "<BR> <B>3.</B> Assume control of the station for the Sol Federation, and initiate evacuation procedures to get non-offending citizens \
+	away from the scene."
+	missiondesc += "<BR> <B>4.</B> Lethal force is authorized at all times."
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/sol_fed.dm:223]: This feels like a bit much for something that's not admin-controlled at all.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm [hunk 7]
```diff
@@ -178,28 +262,37 @@ GLOBAL_LIST_INIT(emergency_responders, list())
...
 		icon_state = "reporter_on"
 		GLOB.votes_for_bad_911_call++
 		to_chat(user, span_warning("You have reported the 911 call as fradulent. \
 		Current Votes: [GLOB.votes_for_bad_911_call]/[GLOB.amt_911_responders] votes."))
-		if(GLOB.votes_for_bad_911_call >= (GLOB.amt_911_responders / 2))
+		if(GLOB.votes_for_bad_911_call >= GLOB.amt_911_responders)
 			to_chat(user, span_warning("The 911 call has been declared fradulent!"))
 			GLOB.fradulent_911_declared = TRUE
 			var/datum/bank_account/station_balance = SSeconomy.get_dep_account(ACCOUNT_CAR)
 			station_balance?._adjust_money(-20000) // paying for the gas to drive all the fuckin' way out to the frontier
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/sol_fed.dm:283]: What happens if the station doesn't have the funds?

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm [hunk 8]
```diff
@@ -224,6 +317,150 @@ GLOBAL_LIST_INIT(emergency_responders, list())
...
+					cop.mind.set_assigned_role(SSjob.GetJobType(ert_antag.ert_job_path))
+					SSjob.SendToLateJoin(cop)
+
+					var/obj/item/gangster_cellphone/phone = new() // biggest gang in the city
+					phone.gang_id = "911"
+					phone.name = "911 branded cell phone"
+					var/phone_equipped = phone.equip_to_best_slot(cop)
+					if(!phone_equipped)
+						to_chat(cop, "Your [phone.name] has been placed at your feet.")
+						phone.forceMove(get_turf(cop))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/sol_fed.dm:426]: These just all feel like they could be simplified using inheritance and variables for the content of those messages, etc.

File: modular_skyrat/modules/goofsec/jobs/department_guards.dm [hunk 9]
```diff
@@ -122,6 +122,7 @@
 
 /datum/id_trim/job/orderly
 	assignment = "Orderly"
+	trim_icon = 'modular_skyrat/master_files/icons/obj/card.dmi'
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/goofsec/jobs/department_guards.dm:125]: This should go in a different PR.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Got a fair bit of cleaning up to do, but we're getting there.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm (hunk 7 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:283]: Caps at 0 credits.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm (hunk 5 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:80]: Admittedly stole it from job code, I can drop the bank account bits.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm (hunk 4 above)
Comment (Iamgoofball) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:8]: These variables gotta go somewhere and they're in a scope that requires them to be global. It's that or stick them on a subsystem.

File: code/game/machinery/computer/communications.dm (hunk 3 above)
Comment (Iamgoofball) [reply code/game/machinery/computer/communications.dm:440]: Sure.

File: code/game/machinery/computer/communications.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/game/machinery/computer/communications.dm:438]: Yeah, it's an oddly formatted proc

File: code/game/machinery/computer/communications.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/game/machinery/computer/communications.dm:425]: I don't want to leave any ambiguity for people on that one.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm (hunk 5 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:80]: Would be preferrable

File: code/game/machinery/computer/communications.dm (hunk 1 above)
Comment (GoldenAlpharex) [reply code/game/machinery/computer/communications.dm:425]: I mean, it can be said differently. I'm sure we can come up with something better.

File: modular_skyrat/modules/goofsec/jobs/sol_fed.dm (hunk 4 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:8]: Surely there's another way around it.

Comment (Iamgoofball) [reply modular_skyrat/modules/goofsec/jobs/sol_fed.dm:8]: It's either storing these all in one mega glob, storing them on a subsystem, or list looping shenanigans.

File: modular_skyrat/modules/goofsec/jobs/department_guards.dm (hunk 9 above)
Comment (ghost) [reply modular_skyrat/modules/goofsec/jobs/department_guards.dm:125]: Doing it RN (God I NEED it)

Comment (ghost) [reply modular_skyrat/modules/goofsec/jobs/department_guards.dm:125]: #9482
